tests/components/mailbox/test_init.py

```diff
@@ -0,0 +1,148 @@
+"""The tests for the mailbox component."""
+import json
+from hashlib import sha1
+import requests
+
+from homeassistant.setup import setup_component
+import homeassistant.components.mailbox as mailbox
+import homeassistant.components.http as http
+
+from tests.common import (
+    get_test_home_assistant, get_test_instance_port, assert_setup_component)
+
+
+class TestMailbox(object):
+    """Test class for mailbox."""
+
+    def setup_method(self):
+        """Initialize platform."""
+        self.hass = get_test_home_assistant()
+
+        setup_component(
+            self.hass, http.DOMAIN,
+            {http.DOMAIN: {http.CONF_SERVER_PORT: get_test_instance_port()}})
```

Comment:
Please do not use `get_test_instance_port`. We deprecated the use of them.

Have a look at the API tests for example how to write tests that consume HTTP views https://github.com/home-assistant/home-assistant/blob/dev/tests/components/test_api.py

That file is also a good example showing how you should not use a test class anymore but instead use functions at the root level to represent tests.